app/scripts/controllers/network/network.js

```diff
@@ -235,6 +251,31 @@ export default class NetworkController extends EventEmitter {
   // Private
   //
 
+  async _checkInfuraAvailability(network) {
+    const rpcUrl = `https://${network}.infura.io/v3/${this._infuraProjectId}`;
+    try {
+      const response = await fetchWithTimeout(rpcUrl, {
+        method: 'POST',
+        body: JSON.stringify({
+          jsonrpc: '2.0',
+          method: 'eth_blockNumber',
+          params: [],
+          id: 1,
+        }),
+      });
+      if (response.ok) {
```

Comment:
We might also want to listen in this function for network changes, and bail here without emitting any event if the network has switched since the function began. Otherwise this might resolve after a subsequent switch, setting this flag on the wrong network.